src/wallet/wallet2.cpp

```diff
@@ -4236,6 +4410,329 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_address_info", request, response, m_http_client, std::chrono::seconds(30), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "get_address_info");
+  return true;
+}
+
+void wallet2::light_wallet_get_address_txs()
+{
+  MDEBUG("Refreshing light wallet");
+  
+  cryptonote::COMMAND_RPC_GET_ADDRESS_TXS::request ireq;
+  cryptonote::COMMAND_RPC_GET_ADDRESS_TXS::response ires;
+  
+  ireq.address = get_account().get_public_address_str(m_testnet);
+  ireq.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_address_txs", ireq, ires, m_http_client, std::chrono::seconds(30), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "get_address_txs");
+  
+  // Store blockchain info
+  m_light_wallet_scanned_block_height = ires.scanned_block_height;
+  m_light_wallet_blockchain_height = ires.blockchain_height;
+  
+  // Create searchable vectors
+  std::vector<crypto::hash> payments_txs;
+  for(const auto &p: m_payments)
+    payments_txs.push_back(p.second.m_tx_hash);
+  std::vector<crypto::hash> unconfirmed_payments_txs;
+  for(const auto &up: m_unconfirmed_payments)
+    unconfirmed_payments_txs.push_back(up.second.m_tx_hash);
+
+
+  uint64_t total_received, total_sent;
+  for (const auto &t: ires.transactions) {
+    total_received = (t.total_received.empty()) ? 0 :  boost::lexical_cast<uint64_t>(t.total_received);
+    total_sent = (t.total_sent.empty()) ? 0 :  boost::lexical_cast<uint64_t>(t.total_sent);
+
+    // Check key image
+    for(const auto &so: t.spent_outputs)
+    {
+      crypto::key_image calculated_key_image;
+      cryptonote::keypair in_ephemeral;
+      crypto::public_key tx_public_key = AUTO_VAL_INIT(tx_public_key);
+      string_tools::hex_to_pod(so.tx_pub_key, tx_public_key);
+
+      cryptonote::generate_key_image_helper(get_account().get_keys(), tx_public_key, so.out_index, in_ephemeral, calculated_key_image);
+
+      // mm key image;
+      crypto::key_image key_image;
+      string_tools::hex_to_pod(so.key_image, key_image);
+
+      if(key_image != calculated_key_image){
+        total_sent -= boost::lexical_cast<uint64_t>(so.amount);
+      }
+    }
+
+    // Do not add tx if empty. 
+    if(total_sent == 0 && total_received == 0)
+      continue;
+    
+    crypto::hash payment_id = null_hash;
+    crypto::hash tx_hash;
+    
+    string_tools::hex_to_pod(t.payment_id, payment_id);
+    string_tools::hex_to_pod(t.hash, tx_hash);
+    // Parse timestamp
+    std::tm tm = {};
+    std::istringstream ss(t.timestamp);
+    uint64_t ts;
+    if (ss >> std::get_time(&tm, "%Y-%m-%dT%H:%M:%S"))
+       ts = std::mktime(&tm);
+    else
+      MERROR("Date parsing error: " << t.timestamp);
```

Comment:
You then want to set ts to some dummy date like the epoch to avoid using uninitialised data on error (or just throw)